Px.Utils/Validation/SyntaxValidation/SyntaxValidation.cs

```diff
@@ -0,0 +1,291 @@
+            foreach (var entry in entries)
+            {
+                foreach (ValidationFunctionDelegate function in validationFunctions)
+                {
+                    ValidationFeedbackItem? feedback = function(entry, syntaxConf);
+                    if (feedback is not null)
+                    {
+                        report.FeedbackItems.Add((ValidationFeedbackItem)feedback);
+                    }
+                }
+            }
+        }
+
+        ///<summary>
+        /// Builds a list of <see cref="KeyValuePairValidationEntry"/> objects from a list of StringValidationEntry objects.
+        ///</summary>
+        /// <param name="stringEntries">The list of <see cref="StringValidationEntry"/> objects to be converted.</param>
+        /// <param name="syntaxConf">The syntax configuration for the PX file.</param>
+        /// <returns>A list of <see cref="KeyValuePairValidationEntry"/> objects built from the provided list of StringValidationEntry objects.</returns>
+        public static List<KeyValuePairValidationEntry> BuildKeyValuePairs(List<StringValidationEntry> stringEntries, PxFileSyntaxConf syntaxConf)
+        {
+            return stringEntries.Select(entry =>
+            {
+                string[] split = entry.EntryString.Split(syntaxConf.Symbols.KeywordSeparator);
+                string value = split.Length > 1 ? split[1] : string.Empty;
+                return new KeyValuePairValidationEntry(entry.Line, entry.Character, entry.File, new KeyValuePair<string, string>(split[0], value));
+            }).ToList();
+        }
+
+        /// <summary>
+        /// Builds a list of <see cref="StructuredValidationEntry"/> objects from a list of <see cref="KeyValuePairValidationEntry"/> objects.
+        /// </summary>
+        /// <param name="keyValuePairs">The list of <see cref="KeyValuePairValidationEntry"/> objects to be converted.</param>
+        /// <param name="syntaxConf">The syntax configuration for the PX file.</param>
+        /// <returns>A list of <see cref="StructuredValidationEntry"/> objects built from the provided list of <see cref="KeyValuePairValidationEntry"/> objects.</returns>
+        public static List<StructuredValidationEntry> BuildStructuredEntries(List<KeyValuePairValidationEntry> keyValuePairs, PxFileSyntaxConf syntaxConf)
+        {
+            return keyValuePairs.Select(entry =>
+            {
+                ValidationEntryKey key = ParseValidationEntryKey(entry.KeyValueEntry.Key, syntaxConf);
+                return new StructuredValidationEntry(entry.Line, entry.Character, entry.File, key, entry.KeyValueEntry.Value);
+            }).ToList();
+        }
+
+        private static List<StringValidationEntry> BuildStringEntries(Stream stream, Encoding encoding, PxFileSyntaxConf syntaxConf, string filename, int bufferSize)
+        {
+            SyntaxValidation.LineCharacterState state = new() { Line = 0, Character = 0 };
+            stream.Seek(0, SeekOrigin.Begin);
```

Comment:
Why seek 0? it should be expected that the parameter given is at the correct position. If some other function reads the stream before this, it should return the stream to the original position. It is not guaranteed that the thing we want to validate starts at 0.